step-automation-packages/step-automation-packages-controller/src/main/java/step/automation/packages/AutomationPackageServices.java

```diff
@@ -118,16 +116,17 @@ public List<String> executeAutomationPackage(@FormDataParam("file") InputStream
     @PUT
     @Path("/{id}")
     @Consumes(MediaType.MULTIPART_FORM_DATA)
-    @Produces(MediaType.TEXT_PLAIN)
+    @Produces(MediaType.APPLICATION_JSON)
     @Secured(right = "automation-package-write")
-    public void updateAutomationPackageById(@PathParam("id") String id,
+    public AutomationPackageUpdateResult updateAutomationPackageById(@PathParam("id") String id,
```

Comment:
Signature change => we should ensure it has no impact on the FE

Reply:
The FE is informed and updating accordingly

Reply:
@david-stephan @jeromecomte 
Does this new optional 'async' parameter really affect FE somehow?

Reply:
@iegorov777 yes we decided the FE would call it in async mode too, so they have to adapt and show when the udpate is delayed.